src/prefect/filesystems.py

```diff
@@ -791,47 +793,103 @@ class GitHub(ReadableDeploymentStorage):
         default=None,
         description="An optional reference to pin to; can be a branch name or tag.",
     )
+    access_token: Optional[SecretStr] = Field(
+        name="Personal Access Token",
+        default=None,
+        description="A github Personal Access Token (PAT).",
+    )
+
+    @validator("access_token")
+    def _ensure_credentials_go_with_https(cls, v: str, values: dict):
+        """Ensure that credentials are not provided with 'SSH' formatted GitHub URLs."""
+        if v is not None:
+            if urllib.parse.urlparse(values["repository"]).scheme != "https":
+                raise InvalidRepositoryURLError(
+                    (
+                        "Crendentials can only be used with GitHub repositories "
+                        "using the 'HTTPS' format. You must either remove the "
+                        "credential if you wish to use the 'SSH' format and are not "
+                        "using a private repository, or you must change the repository "
+                        "url to the 'HTTPS' format. "
+                    )
+                )
+
+        return v
+
+    def _create_repo_url(self) -> str:
```

Comment:
Should this be a `repository` validator?

Reply:
I figured this way keeps us from having an attribute that exposes the secret value.